tools/python/boutiques/bosh.py

```diff
@@ -99,20 +99,25 @@ def execute(*params):
         parser.add_argument("-s", "--stream", action="store_true",
                             help="Streams stdout and stderr in real time "
                             "during execution.")
+        parser.add_argument("-z", "--zenodo", action="store_true",
+                            help="Download and use a descriptor published "
+                            "on Zenodo.")
         results = parser.parse_args(params)
         descriptor = results.descriptor
         inp = results.invocation
 
         # Validate invocation and descriptor
-        valid = invocation(descriptor, '-i', inp)
+        if not results.zenodo:
```

Comment:
this should actually have a TODO on it. I need to allow `bosh validate` to use a descriptor from Zenodo as well.

Reply:
Pretty much all the functions have to accept Zenodo IDs then. This behavior could be factorized in a function that would accept a file, a JSON string, or a Zenodo id, and return a JSON object or dictionary representing the descriptor. All the methods would call this function to parse the descriptor or invocation. Then we could easily add more representations or behavior in the future, for instance caches, etc.

Reply:
I already have the function that checks if the descriptor is a JSON file or object, so I can just extend that further to check if it's a Zenodo ID.